assets/js/modules/analytics-4/datastore/custom-dimensions.js

```diff
@@ -195,6 +199,70 @@ const baseActions = {
 			payload: { customDimensions: [] },
 		};
 	},
+
+	/**
+	 * Sets a schedule timeout ID for syncing available custom dimensions in state.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {number} syncTimeoutID The timeout ID.
+	 * @return {Object} A redux-style action.
+	 */
+	setSyncTimeoutID( syncTimeoutID ) {
+		return {
+			payload: { syncTimeoutID },
+			type: SET_SYNC_TIMEOUT_ID,
+		};
+	},
+
+	/**
+	 * Schedules a sync of available custom dimensions in state.
+	 *
+	 * @since n.e.x.t
+	 */
+	*scheduleSyncAvailableCustomDimensions() {
+		yield {
+			payload: {},
+			type: SCHEDULE_SYNC_AVAILABLE_CUSTOM_DIMENSIONS,
+		};
+	},
+};
+
+export const baseControls = {
+	[ SCHEDULE_SYNC_AVAILABLE_CUSTOM_DIMENSIONS ]: createRegistryControl(
+		( { select, dispatch } ) =>
+			() => {
+				const syncTimeoutID =
+					select( MODULES_ANALYTICS_4 ).getSyncTimeoutID();
+				const isSyncingAvailableCustomDimensions =
+					select(
+						MODULES_ANALYTICS_4
+					).isFetchingSyncAvailableCustomDimensions();
+
+				if ( !! syncTimeoutID ) {
+					clearTimeout( syncTimeoutID );
+					dispatch( MODULES_ANALYTICS_4 ).setSyncTimeoutID(
+						undefined
+					);
+				}
+
+				if ( isSyncingAvailableCustomDimensions ) {
+					return;
+				}
+
+				const timeoutID = setTimeout( async () => {
+					await dispatch(
+						MODULES_ANALYTICS_4
+					).fetchSyncAvailableCustomDimensions();
+
+					dispatch( MODULES_ANALYTICS_4 ).setSyncTimeoutID(
+						undefined
+					);
```

Comment:
I'm not sure this needs to be unset aside from the above where it calls `clearTimeout`. Since the timeout is really a side-effect specific to the control, the reference to it doesn't even need to be in the store I don't think (it could just be a private variable in scope here).

Reply:
> Since the timeout is really a side-effect specific to the control, the reference to it doesn't even need to be in the store I don't think (it could just be a private variable in scope here).

Okay, I tried doing this with a scoped variable and also using lodash's `debounce` function. It seems the timeout ID isn't found across multiple dispatches, and the fetch is run as many times as the action is dispatched, just with the 2-second delay. Any ideas?

> I'm not sure this needs to be unset aside from the above where it calls `clearTimeout`.

Since we're using the existing pattern of keeping the timeout ID in the state (for now, unless we find a better solution), I think it might be worth unsettling this here just to make sure `clearTimeout` isn't unnecessarily called on the next dispatch. WDYT?